src/pages/EnablePayments/AdditionalDetailsStep.js

```diff
@@ -47,68 +55,122 @@ class AdditionalDetailsStep extends React.Component {
+            legalFirstName: lodashGet(props.walletAdditionalDetailsDraft, 'legalFirstName', ''),
+            legalMiddleName: lodashGet(props.walletAdditionalDetailsDraft, 'legalMiddleName', ''),
+            legalLastName: lodashGet(props.walletAdditionalDetailsDraft, 'legalLastName', ''),
+            addressStreet: lodashGet(props.walletAdditionalDetailsDraft, 'addressStreet', ''),
+            addressCity: lodashGet(props.walletAdditionalDetailsDraft, 'addressCity', ''),
+            addressState: lodashGet(props.walletAdditionalDetailsDraft, 'addressState', ''),
+            addressZip: lodashGet(props.walletAdditionalDetailsDraft, 'addressZip', ''),
+            phoneNumber: lodashGet(props.walletAdditionalDetailsDraft, 'phoneNumber', ''),
+            dob: lodashGet(props.walletAdditionalDetailsDraft, 'dob', ''),
+            ssn: lodashGet(props.walletAdditionalDetailsDraft, 'ssn', ''),
         };
     }
 
+    /**
+     * @param {String} fieldName
+     * @returns {String}
+     */
+    getErrorText(fieldName) {
+        const errors = lodashGet(this.props.walletAdditionalDetails, 'errorFields', {});
+        if (!errors[fieldName]) {
+            return '';
+        }
+
+        return `${this.props.translate(this.fieldNameTranslationKeys[fieldName])} ${this.props.translate('common.isRequiredField')}.`;
+    }
+
+    /**
+     * @returns {Boolean}
+     */
+    validate() {
+        const errors = {};
+
+        if (!ValidationUtils.isValidPastDate(this.state.dob)) {
+            errors.dob = true;
+        }
+
+        if (!ValidationUtils.isValidAddress(this.state.addressStreet)) {
+            errors.addressStreet = true;
+        }
+
+        if (!ValidationUtils.isValidSSNLastFour(this.state.ssn)) {
+            errors.ssn = true;
+        }
+
+        _.each(this.requiredFields, (requiredField) => {
+            if (ValidationUtils.isRequiredFulfilled(this.state[requiredField])) {
+                return;
+            }
+
+            errors[requiredField] = true;
+        });
+
+        Wallet.setAdditionalDetailsErrors(errors);
+        return _.size(errors) === 0;
+    }
+
+    activateWallet() {
+        Wallet.setAdditionalDetailsStep(false);
+
+        if (!this.validate()) {
+            return;
+        }
+
+        BankAccounts.activateWallet(CONST.WALLET.STEP.ADDITIONAL_DETAILS, {
+            personalDetails: this.state,
+        });
+    }
+
+    /**
+     * @param {String} fieldName
+     * @param {String} value
+     */
+    clearErrorAndSetValue(fieldName, value) {
+        this.setState({[fieldName]: value});
+        Wallet.updateAdditionalDetailsDraft({[fieldName]: value});
+        const errors = lodashGet(this.props, 'walletAdditionalDetails.errorFields', {});
+        if (!lodashGet(errors, fieldName, false)) {
+            return;
+        }
+
+        const newErrors = lodashCloneDeep(errors);
+        lodashUnset(newErrors, fieldName);
+        Wallet.setAdditionalDetailsErrors(newErrors);
```

Comment:
I thought about maybe taking the time to abstract some of the logic out of `ReimbursementAccountUtils` to reuse here, but decided not to since this will all be replaced eventually by our new forms system whenever it is ready.

https://github.com/Expensify/App/blob/9f2b4f025961d5513dcf949b896f3d0fef956a98/src/libs/ReimbursementAccountUtils.js#L32-L43

I will create a follow up to address this if it looks like the forms project will take a long time.